impress-org/givewp
# Donor Profile Verify Email now supports reCAPTCHA

Resolves #5637

## Description



Validates a Google reCAPTCHA token on the VerifyEmail route, if reCAPTCHA is enabled for Access Control.

## Affects



- Adds `react-google-recaptcha` as a dependency.
- Fixes event handling in the Verify Email form.
- Passes a reCAPTCHA token to the `verifyEmailWithAPI` utility.

## Pre-review Checklist




## Testing Instructions



In Settings -> General -> Access Control

- Enable Access Control
- Enable reCAPTCHA 
- Set reCAPTCHA Site Key
- Set reCAPTCHA Secret Key

With an unauthenticated session...
With the Donor Profile block on the page...

- Input an email address belonging to a WP User.
- Click the "Verify Email" button
